### PR TITLE
Work around keyword name collisions in Go discovery sample gen

### PR DESCRIPTION
**Pull request by tcoffee-google**
Temporary fix pending general collision handling in MVVM refactoring.

### CODE REVIEW

**Comment by tcoffee-google** [comment]
@saicheems note this provides the reserved word set for Go we will need for MVVM refactoring.

File: src/main/java/com/google/api/codegen/util/go/GoTypeTable.java [hunk 1]
```diff
@@ -104,4 +105,38 @@ private static boolean isStandardImport(String importPath) {
     // we might have to special case them.
     return !importPath.contains("/");
   }
+
+  /**
+   * A set of Go reserved words. See
+   * https://golang.org/ref/spec#Keywords
+   */
+  public static final ImmutableSet<String> KEYWORD_BUILT_IN_SET =
```
Comment (saicheems) [inline src/main/java/com/google/api/codegen/util/go/GoTypeTable.java]: Any chance we can call this `RESERVED_KEYWORD_SET`?
I know `KEYWORD_BUILT_IN_SET` is the same naming as in `NodeJSTypeTable`, but I think the above name is a bit more descriptive for this set.

I wonder if it's having another set or including non-reserved identifiers like type names. While we technically don't have to, we could run into some hairy situations like:
```
int := 3
```

Comment (saicheems) [inline src/main/java/com/google/api/codegen/util/go/GoTypeTable.java]: Or maybe `RESERVED_KEYWORD_SET`

Comment (saicheems) [inline src/main/java/com/google/api/codegen/util/go/GoTypeTable.java]: Though `RESERVED_IDENTIFIER_SET` is a more accurate name if we do decide to include the names in builtin:

https://golang.org/pkg/builtin/

All of these can be overridden by declaring a var of the same name

**Comment by saicheems** [comment]
The mechanism lgtm

File: src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java [hunk 2]
```diff
@@ -46,6 +47,10 @@ public String getMethodName(Method method) {
     return lowerCamelToUpperCamel(super.getMethodName(method));
   }
 
+  public boolean isKeyword(String name) {
```
Comment (saicheems) [inline src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java]: isReserved may be a better function name

**Comment by saicheems** [comment]
LGTM

**Comment by vchudnov-g** [comment]
I just ran this against the sheets API, and the Go compilecheck passes.

LGTM.

File: src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java (hunk 2 above)
Comment (tcoffee-google) [inline src/main/java/com/google/api/codegen/go/GoDiscoveryContext.java]: Done

**Pull request merged**